src/FAudio_gstreamer.c

```diff
@@ -473,15 +542,42 @@ uint32_t FAudio_GSTREAMER_init(FAudioSourceVoice *pSourceVoice, uint32_t type)
 		gst_init(NULL, NULL);
 	}
 
+	if (type == FAUDIO_FORMAT_XMAUDIO2 && !FAudio_GSTREAMER_XMA2_Mimetype)
+	{
+		/* Old versions ship with unknown/unknown as the sink caps mimetype.
+		 * A patch has been submitted (and merged!) to expose avdec_xma2 as audio/x-xma with xmaversion 2:
+		 * https://gitlab.freedesktop.org/gstreamer/gst-libav/-/merge_requests/124
+		 * For now, try to find avdec_xma2 if it's found, match the mimetype on the fly.
+		 * This should also take future alternative XMA2 decoders into account if avdec_xma2 is missing.
```

Comment:
May be good to add a note for what we can discard later on when we can assume that the patch is in active use.

Reply:
The discardable stuff can now be skipped by defining `FAUDIO_GST_LIBAV_EXPOSES_XMA2_CAPS_IN_CURRENT_YEAR`, hopefully that's good enough until the time comes to make that the default 😅